README.md
# News-Central
A proper PHP-MySQL Login-Logout System with Registration and Live News Feed scraped via Python Script.


news.txt
1. Jalaluddin Haqqani, founder of militant Haqqani network, dead: Afghan Taliban
2. Afghan Taliban Announces Death Of Haqqani Network Founder
3. Founder of militant Afghan Haqqani network dies: Taliban
4. Afghan Taliban announces death of Haqqani network leader
5. How CIA nutured Jalaluddin Haqqani as an asset?
6. Brazil's iconic museum gutted: A look at the treasures it once housed - Rio museum gutted in fire
7. 90% of 20 million relics may have been damaged after fire at Rio museum: Officials
8. Brazil’s 200-year-old National Museum reduced to ashes
9. Outrage over fire at Brazil national museum | Al Jazeera English
10. Novelist Paulo Coelho on Brazil's National Museum fire: 'The country is in tears'
11. What Nirmala Sitharaman, Sushma Swaraj Will Negotiate With US Officials
12. India asks US to not forget 'strategic relations' amid sanction threat
13. Talks on for logistics deal with Russia
14. India-US 2+2 talks: Focus on concrete outcomes; New Delhi should get over Cold War hangover
15. '2+2 dialogue': India, US to discuss sale of drones, exchange of satellite data
16. First jolt to Imran Khan as top aide quits after corruption case filed
17. Scientist Who Developed North Korea's Missiles, Nuclear Weapons Dies
18. Japan hit by strongest typhoon in 25 years; nearly 600 flights cancelled
19. Strongest typhoon in quarter century hits Japan - Typhoon Jebi: Strongest in 25 years
20. Millions asked to evacuate as powerful typhoon Jebi pounds Japan
21. Typhoon Jebi Makes Landfall In Japan, 500 Flights Cancelled
22. Japan issues evacuation advisories as typhoon Jebi approaches
23. China kindergarten principal sacked over pole dance show at school
24. Chinese School Apologises For Welcoming Kids (And Shocked Parents) With Pole Dance
25. BIZARRE! Kindergarten principal in China permits pole dance in school, gets fired – Watch video
26. Kindergarten principal fired over `strip-like pole dance` in school
27. Kindergarten principal in China Fired for Allowing Pole Dance to Welcome Students
28. India to use Iranian ships to skirt US oil sanctions
29. US Chokes Iran Oil Flows To Buyers Who Vowed Resistance, Including India
30. India, China face mounting pressure from US; Trump says stop buying oil from Iran or face sanctions
31. India allows state refiners to use Iran tankers, insurance for oil imports
32. Secret oil shipments could help Iran cushion US sanctions blow
33. British PM Theresa May's Brexit blueprint blasted by Boris Johnson
34. Johnson steps up criticism of May’s Brexit plan
35. Indian-Origin UK Lawmaker Challenges Theresa May's Brexit Strategy
36. Boris Johnson’s latest Brexit outburst combines madness and mendacity
37. Pressure mounts on UK PM Theresa May as critics tear into her Brexit plan
38. Indian diplomats likely to get approval for membership to Pakistan's Islamabad Club by Imran Khan government
39. Pakistan may approve club membership for Indian diplomats
40. Pak Disagrees With US, Says India Has "No Role To Play" In Afghanistan
41. India must not take China's declaration of friendship..ce value and risk slowing improvement of ties with US
42. Choppy waters ahead as US seeks to ramp up Pakistan pressure
43. India, Cyprus vow to curb money laundering
44. India is world's fastest-growing economy, says President Kovind in Cyprus
45. Pak At Risk Of Debt Over Trade Corridor Project With China: US Senators
46. Pakistani ASF female officer penalised for lip-sync video; netizens angry
47. Pakistan woman lip-syncs Indian song, penalised by authorities after video goes viral
48. Pakistan Airport Worker Penalised after Video of Her Lip-Syncing Indian Song Goes Viral
49. Pakistan Airport Security Force staffer lip-syncs to an Indian song; gets penalized
50. Pakistani Woman Lip-Synced Indian Song; Loses Perks For 2 Years
51. NIA court frames charges against six alleged ISIS members
52. 6 ISIS Operatives Charged For Plotting Terror Attacks In Kerala
53. Lessons from Argentina: How to Prevent Financial Crises in Emerging Markets?
54. Argentina's Macri raises taxes, axes ministries after peso plunge | Al Jazeera English
55. Israel Signals To Attack Iranian Weaponry In Iraq
56. Israel signals it could target Iranian military assets in Iraq
57. Israel Signals It Could Attack Iranian Weaponry In Iraq
58. Report On Moving Missiles From Iran To Iraq "Without Evidence": Baghdad
59. Death toll in Libya's Tripoli rises as fighting rages on
60. In Libya, Facebook is used to trade arms and find targets
61. Hundreds of prisoners escape from Tripoli jail amid militia fighting
62. Israelis protest Philippine President Duterte
63. Yup, It's True. Space Elevator To Be Tested This Month
64. Going up! Japan to test mini 'space elevator'
65. 2+2 talks: In US courtship of India and separation fr..kistan, a lesson from Cold War and revenge of history
66. Imran Khan government faces first challenge as anti-US sentiment rises in Pakistan after aid cut
67. Pressing Pakistan To Indiscriminately Target Terrorist Groups: US
68. $300 million not aid, but what US owes to Pakistan for support in war against terror, says Shah Mehmood Qureshi
69. 'Aid cut by US will push Pak closer to China’
70. Imran Khan's close aide Dr Arif Alvi elected Pakistan's 13th president: Report
71. Pakistan presidential polls 2018: Islamabad lawmakers vote to elect new President
72. Pakistan Tehreek-e-Insaf Party Member Arif Alvi Elected New President
73. Pakistani lawmakers begin electing new president
74. Imran Khan's close ally Arif Alvi elected new Pakistan president
75. Trump threatens to exclude Canada from new Nafta deal
76. US ratchets up pressure on Canada, Mexico, trade partners
77. Donald Trump Threatens To Exclude Canada From New NAFTA Deal
78. 'No political necessity' to keep Canada in new NAFTA deal: US President Donald Trump
79. India in billion-dollar initiative to build world's biggest statues

dbat.php
<?php
include ("config.php");
function redirect($url) {
    ob_start();
    header('Location: '.$url);
    ob_end_flush();
    die();
}
$conn = mysql_connect($db_host, $db_user, $db_pass)
or die ("Could not connect to mysql because ".mysql_error());
$createdb = "create database $db_name";
mysql_query($createdb)
or die ("Could not create database because ".mysql_error());
mysql_select_db($db_name)
or die ("Could not select database because ".mysql_error());
$createtab = "create table $db_table (id smallint(5) NOT NULL auto_increment, firstname VARCHAR(30), lastname VARCHAR(30), email VARCHAR(50), gender VARCHAR(10), username VARCHAR(30), password VARCHAR(30), PRIMARY KEY (id), UNIQUE KEY username (username));";
mysql_query($createtab)
or die ("Could not create tables because ".mysql_error());
$url="register_r.html";
redirect($url);
?>

login.php
<?php
include("config.php"); 
function redirect($url) {
    ob_start();
    header('Location: '.$url);
    ob_end_flush();
    die();
}
$link = mysql_connect($db_host, $db_user, $db_pass)
or die ("Could not connect to mysql because ".mysql_error());
mysql_select_db($db_name)
or die ("Could not select database because ".mysql_error());
$match = "select id from $db_table where username = '".$_POST['username']."'
and password = '".$_POST['password']."';"; 
$qry = mysql_query($match)
or die ("Could not match data because ".mysql_error());
$num_rows = mysql_num_rows($qry); 
if ($num_rows <= 0) { 
echo "Sorry, there is no username $username with the specified password.<br>";
echo "<a href=login.html>Try again</a>";
exit; 
} else {
session_start();
$username=$_POST['username'];
$_SESSION['login_user']=$username; 
$url="main_page.php";
redirect($url);
}
?>



get_news.py
import urllib2
from bs4 import BeautifulSoup
url="https://news.google.com/topics/CAAqJggKIiBDQkFTRWdvSUwyMHZNRGx1YlY4U0FtVnVHZ0pKVGlnQVAB?hl=en-IN&gl=IN&ceid=IN%3Aen"
page=urllib2.urlopen(url)
soup=BeautifulSoup(page, 'html.parser')
headlines=[]
null=0
material=soup.find_all("span")
for content in material:
	headline=content
	string=str(content)
	if ('aria-hidden=') and ('class=') not in string:
		for news in headline:
			try:
				headlines.append(news)
			except:
				UnicodeEncodeError,UnicodeDecodeError,UnicodeTranslateError
file=open("news.txt", "w")
count=1
for headline in headlines:
	try:
		headline=headline.encode("utf-8")
		file.write(str(count)+'. '+headline+'\n')
	except:
		UnicodeEncodeError,UnicodeDecodeError,UnicodeTranslateError
	count+=1
file.close()







register.php
<?php 
include("config.php"); 
function redirect($url) {
    ob_start();
    header('Location: '.$url);
    ob_end_flush();
    die();
}
$link = mysql_connect($db_host, $db_user, $db_pass)
or die ("could not connect to mysql because ".mysql_error());
mysql_select_db($db_name)
or die ("could not select database because ".mysql_error());
$check = "select id from $db_table where username = '".$_POST['username']."';";
$qry = mysql_query($check) or die ("could not match data because ".mysql_error());
$num_rows = mysql_num_rows($qry); 
if ($num_rows != 0) { 
echo "sorry, the username $username is already taken.<br>";
echo "<a href=register.html>Try again</a>";
exit; 
} else {
$insert = mysql_query("insert into $db_table values ('NULL',
'".$_POST["firstname"]."',
'".$_POST["lastname"]."',
'".$_POST["email"]."',
'".$_POST["gender"]."',
'".$_POST["username"]."',
'".$_POST["password"]."')")
or die("could not insert data because ".mysql_error());
$url="login_r.html";
redirect($url);
}
?>

main_page.php
<html>
<head>
<meta charset="UTF-8"></meta>
<title><?php session_start(); print_r ($_SESSION['login_user']);?>-NC</title>
<style>
div {
    width: 50px;
    height: 50px;
    background: red;
    position: relative;
    animation: mymove 5s infinite;
}
@keyframes mymove {
    from {left: 100px;}
    to {left: 1050px;}
    0%   {background: red;}
    25%  {background: yellow;}
    50%  {background: blue;}
    75%  {background: green;}
    100% {background: red;}
}
p.bottom{
	text-align: center; bottom: 30px; left: 460px;
}
body{
	background-color: powderblue;
}
</style>
</head>
<body>
<h1 align="center">NEWS CENTRAL</h1>
<h2 align="center">Your daily dose of International News</h2>
<table>
<col width="100">
<col width="1050">
<col width="80">
<tr>
<td><p>Welcome <b><?php print_r ($_SESSION['login_user']);?></b>, </p></td>
<td colspan="10"><div></div></td>
<td><form action="logout.php" method="post" align="right"><input type="submit" name="logout" value="LOGOUT"></input></form></td>
</tr>
</table>
<i><b><p>Here is the list of all the latest news around the globe.</p></b></i><br>
<?php 
ini_set('max_execution_time', 300);
exec("python get_news.py"); 
$file=fopen("news.txt", "r");
while(!feof($file))
  {
  echo "<hr>" .mb_convert_encoding(fgets($file),"HTML-ENTITIES","UTF-8"). "<br>";
  }
fclose($file);
?>
<br>
<i><p>*You are all Up-To-Date for now, Thank You for using News Central.*</p></i><br>
<p class="bottom">©2018|Created and Maintained by Aaditya Vaidya|All Rights Reserved</p>
</body>
</html>